MapboxCoreNavigation/RouteController.swift

```diff
@@ -444,3 +532,121 @@ extension RouteController: CLLocationManagerDelegate {
         incrementRouteProgress(alertLevel, location: location, updateStepIndex: updateStepIndex)
     }
 }
+
+struct SessionState {
+    let identifier = UUID()
+    var departureTimestamp: Date?
+    var arrivalTimestamp: Date?
+    
+    var totalDistanceCompleted: CLLocationDistance = 0
+    
+    var numberOfReroutes = 0
+    var lastReroute: Date?
+    
+    var currentRoute: Route!
+    var currentRequestIdentifier: String?
+    
+    var originalRoute: Route!
+    var originalRequestIdentifier: String?
+    
+    var pastLocations = FixedLengthBuffer<CLLocation>(length: 40)
+}
+
+// MARK: - Telemetry
+extension RouteController {
+    
+    func sendDepartEvent() {
+        let eventName = "navigation.depart"
+        
+        NSLog("Sending \(eventName)")
+        
+        var eventDictionary = MGLMapboxEvents.addDefaultEvents(routeController: self)
+        eventDictionary["event"] = eventName
+
+        MGLMapboxEvents.pushEvent(eventName, withAttributes: eventDictionary)
+        MGLMapboxEvents.flush()
+    }
+    
+    func sendFeedbackEvent(event: CoreFeedbackEvent) {
+        // remove from outstanding event queue
+        if let index = outstandingFeedbackEvents.index(of: event) {
+            outstandingFeedbackEvents.remove(at: index)
+        }
+        
+        let eventName = event.eventDictionary["event"] as! String
+        
+        NSLog("Sending \(eventName)")
```

Comment:
Remove any print statements before merging. If these print statements are really needed for debugging, we should make them debug-only, either using a compiler flag or perhaps a user default.